tests/integration/test_auth.py

```diff
@@ -79,21 +79,20 @@ def test_auth_can_create_authenticated_requests_sessions():
     assertions.assertTrue("Bearer" in session.headers["Authorization"])
 
 
-def test_auth_can_fetch_s3_credentials():
+@pytest.mark.parametrize("daac", earthaccess.daac.DAACS)
+def test_auth_can_fetch_s3_credentials(daac):
     activate_environment()
     auth = earthaccess.login(strategy="environment")
     assertions.assertTrue(auth.authenticated)
-    for daac in earthaccess.daac.DAACS:
-        if len(daac["s3-credentials"]) > 0:
-            try:
-                logger.info(f"Testing S3 credentials for {daac['short-name']}")
-                credentials = earthaccess.get_s3_credentials(daac["short-name"])
-                assertions.assertIsInstance(credentials, dict)
-                assertions.assertTrue("accessKeyId" in credentials)
-            except Exception as e:
-                logger.error(
-                    f"An error occured while trying to fetch S3 credentials for {daac['short-name']}: {e}"
-                )
+    try:
+        logger.info(f"Testing S3 credentials for {daac['short-name']}")
+        credentials = earthaccess.get_s3_credentials(daac["short-name"])
+        assert isinstance(credentials, dict)
+        assert "accessKeyId" in credentials
+    except Exception as e:
+        logger.error(
+            f"An error occured while trying to fetch S3 credentials for {daac['short-name']}: {e}"
+        )
```

Comment:
I removed the call to `logger.info` because the relevant information will now appear within the name of the test instance due to parametrization (because pytest will construct a unique name as a combination of the test function name _plus_ the value of the parameter), so the log message becomes redundant.

Further, I moved the assertions outside of the `try` block because we do _not_ want to catch assertion errors, otherwise this test might as well not exist since it would otherwise _never_ fail, thus hiding possible bugs. I've also reduced the scope to catch only `RequestException`s, for the same reason.

You must add an import as well: `import requests`

```suggestion
    try:
        credentials = earthaccess.get_s3_credentials(daac["short-name"])
    except requests.RequestException as e:
        logger.error(f"Failed to fetch S3 credentials: {e}")
    else:
        assert isinstance(credentials, dict)
        assert "accessKeyId" in credentials
```

We should likely be even more specific about what we catch. If the exception is more specifically a `requests.HTTPError` with a status code < 500, then we should reraise it because that very likely indicates a bug in our code (as it is a client error, not a server error). However, that is perhaps beyond the scope of this issue, which is to simply parametrize this test.